components/upload-progress-fullpage.tsx

```diff
@@ -74,11 +77,53 @@ const UploadProgressFullpage: React.FC<Props> = ({ file }) => {
     }
   }, [upload]);
 
+
+  const startFileValidation = (_file: File) => {
+    // Attempt to load the file as a video element and inspect its duration
+    // metadata. This is not an authoritative check of video duration, but
+    // rather intended to serve as just a simple and fast sanity check.
+    if (!_file.type.includes("video")) {
+      console.warn(`file type (${_file.type}) does not look like video!`);
+      setInputValidated(true);
+      return;
+    }
+
+    const video = document.createElement('video');
+    video.preload = 'metadata';
+    video.onloadedmetadata = function() {
+      URL.revokeObjectURL(video.src);
+      if (video.duration > MAX_VIDEO_DURATION_SEC) {
+        setValidationError(`file duration (${video.duration.toString()}s) exceeds allowed maximum (${MAX_VIDEO_DURATION_SEC}s)!`);
```

Comment:
I think the error message should be more friendly here 3600 seconds is a bit difficult to understand, '60 minutes' would be better.